src/warning.rs

```diff
@@ -0,0 +1,62 @@
+#[derive(Clone)]
+pub struct Warning {
+    text: Option<String>,
+    priority: i32,
+}
+
+/// Struct for storing and setting warnings (not errors) to be shown to the user.
```

Comment:
These doc-comment describing the struct now show up under the implementation instead of the struct itself, which does not seem intentional to me